src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,163 @@
+            result = 31 * result + (value == null ? 0 : value.hashCode());
+            return result;
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o == null) {
+                return false;
+            }
+            if (o == this) {
+                return true;
+            }
+            if (getClass() != o.getClass()) {
+                return false;
+            }
+            Node<K, V> node = (Node<K, V>) o;
+            return (key == node.key || key != null && key.equals(node.key))
+                    && (value == node.value || value != null && value.equals(node.value));
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || table.length == 0) {
+            table = transfer();
+        }
+        Node<K, V> first;
+        if ((first = table[getIndex(key)]) == null) {
+            table[getIndex(key)] = new Node<>(hash(key), key, value, null);
+        } else {
+            if (first.hash == hash(key)
+                    && (first.key == key || (key != null && key.equals(first.key)))) {
+                first.value = value;
+                return;
+            }
+            Node<K, V> e;
+            Node<K, V> tail;
+            if ((e = first.next) != null) {
+                do {
+                    tail = e;
+                    if (e.hash == hash(key)
+                            && (e.key == key || (key != null && key.equals(e.key)))) {
+                        e.value = value;
+                        return;
+                    }
+                } while ((e = e.next) != null);
+                tail.next = new Node<>(hash(key), key, value, null);
+            } else {
+                first.next = new Node<>(hash(key), key, value, null);
+            }
+        }
+        size++;
+        if (size > threshold) {
+            transfer();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> e;
+        return (e = getNode(key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : key.hashCode();
+    }
+
+    private Node<K,V> getNode(K key) {
+        Node<K,V> first;
+        if (table != null && (table.length) > 0 && ((first = table[getIndex(key)]) != null)) {
+            if (first.hash == hash(key)
+                    && (first.key == key || (key != null && key.equals(first.key)))) {
+                return first;
+            }
+            Node<K, V> e;
+            if ((e = first.next) != null) {
+                do {
+                    if (e.hash == hash(key)
+                            && (e.key == key || (key != null && key.equals(e.key)))) {
+                        return e;
+                    }
+                } while ((e = e.next) != null);
+            }
+        }
```

Comment:
rewrite your logic, it's too complex. An algorithm should be like this:
1. Find bucket by key.
2. create a while loop with such condition (node != null)
3. inside of while check if keys equal, in case it's true just return a value